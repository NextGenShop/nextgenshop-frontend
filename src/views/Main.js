import React from "react";
import CashierMain from "./cashier/CashierPage";
import ShopperMain from "./shopper/ShopperPage";
import Navbar from "../components/Navbar";
import { Redirect } from "react-router-dom";
import { connect } from "react-redux";

function Main({ auth }) {
  if (auth && auth.isAuthenticated && auth.user) {
    const RenderComponent = auth.user.type === "cashier" ? <CashierMain /> : <ShopperMain />;
    return (
      <React.Fragment>
        <Navbar />
        {RenderComponent}
      </React.Fragment>
    );
  } else {
    return <Redirect to="/login" />;
  }
}

const mapStateToProps = (state) => ({
  auth: state.auth,
});

export default connect(mapStateToProps, null)(Main);
