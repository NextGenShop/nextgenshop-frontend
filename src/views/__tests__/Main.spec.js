import React from 'react';
import { shallow } from 'enzyme';
import { Main } from '../Main';
import CashierMain from '../cashier/CashierPage';
import ShopperMain from '../shopper/ShopperPage';

jest.mock('../cashier/CashierPage');
jest.mock('../shopper/ShopperPage');

describe('Main', () => {
  const mockShopper = { type: 'shopper' };
  const mockCashier = { type: 'cashier' };
  test('should correctly render', () => {
    const component = shallow(<Main />);
    expect(component).toMatchSnapshot();
  });
  test('should correctly render with props', () => {
    const props = {
      auth: { user: mockShopper, isAuthenticated: false },
    };
    const component = shallow(<Main {...props} />);
    expect(component).toMatchSnapshot();
  });
  test('should correctly render shopper main', () => {
    const props = {
      auth: { user: mockShopper, isAuthenticated: true },
    };
    const component = shallow(<Main {...props} />);
    expect(component.find(ShopperMain)).toHaveLength(1);
  });
  test('should correctly render cashier main', () => {
    const props = {
      auth: { user: mockCashier, isAuthenticated: true },
    };
    const component = shallow(<Main {...props} />);
    expect(component.find(CashierMain)).toHaveLength(1);
  });
});
