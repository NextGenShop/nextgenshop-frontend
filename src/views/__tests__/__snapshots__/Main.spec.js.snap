// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should correctly render 1`] = `
<Redirect
  to="/login"
/>
`;

exports[`Main should correctly render with props 1`] = `
<Redirect
  to="/login"
/>
`;
